### ISSUE TITLE
Add index for AR_TRN_ARC.IVC_CD

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#258]
_Imported from trac ticket 258.
Created by **kbower**
Opened in trac: Aug 19, 2010, 10:17:13 AM
Last modified in trac: Feb 19, 2011, 9:34:41 PM_
***
# email

From: Kent Bower [mailto:kb@retailarchitects.com]
Sent: Wednesday, August 18, 2010 9:27 AM
To: Jesus Capo; Wayne Liburd; Mariam Farach
Cc: rn@retailarchitects.com
Subject: Re: Querying transactions for a given order.

 

Thank you both for your input.

It sounds like we could safely only display transactions for an order if they are for the given customer, but the followup question I'd like your input for:

Regardless of how GERS does this currently, which way would you like it implemented?
When you click on an order to see its transactions:

    * Show all transactions against that order.

OR

    * Show transactions against that order if from the same customer account as order. (Like gers OCINQ)

We can implement this either way.

Votes?

## JRs response

I would vote to display  ALL TRANSACTIONS against that order. 

No matter where they came from.

 

JR

 
## Personally, I agree with this approach.
However, performance is currently very poor when selecting order transactions in customer history service call unless customer is also joined, so we need an index (at least on ar_trn_arc, maybe also on ar_trn)

**Comment by kentbower** [comment]
_**kbower** trac comment on Aug 20, 2010, 11:59:44 AM:_

Note: added this index manually on Kane's.  It greatly improved performance (apparently no need for one on ar_trn also):

```
SQL> select INDEX_OWNER, INDEX_NAME, TABLE_OWNER, TABLE_NAME, COLUMN_NAME from dba_ind_columns where table_name = 'AR_TRN_ARC' and index_name = 'ARTRNARC_IVCCD_IND';

INDEX_OWNER                    INDEX_NAME                     TABLE_OWNER
------------------------------ ------------------------------ ------------------------------
TABLE_NAME
------------------------------
COLUMN_NAME
-------------------------------------------------------------------------------------------------------------------
ARC                            ARTRNARC_IVCCD_IND             AR
AR_TRN_ARC
IVC_CD
```
However, this needs to be automated for setup-app.

**Comment by kentbower** [comment]
_**kbower** trac comment on Sep 8, 2010, 11:06:21 AM:_

```
create index ARTRNARC_IVCCD_IND on AR_TRN_ARC(IVC_CD);
```

**Comment by kentbower** [comment]
_**rneville** trac comment on Dec 14, 2010, 8:42:11 AM:_

create index artrn_cust_ind on ar_trn(cust_cd);

is also a REALLY good idea.

**Comment by kentbower** [comment]
_**kbower** trac comment on Feb 19, 2011, 9:34:41 PM:_

(Changeset [[1220]](../commit/3b393278db3bb98ecd9652200271f223e5e04514))
 * Add automation for adding indexes.  (closes #258)

**Comment by kentbower** [comment]
_**kbower** trac status change on Feb 19, 2011, 9:34:41 PM:_ **closed**
